dist/index.js

```diff
@@ -40,81 +40,84 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
+    var buffer, check, type, sectorSize, index;
+    return __generator(this, function (_a) {
+        switch (_a.label) {
+            case 0:
+                if (!(input instanceof Uint8Array || input instanceof ArrayBuffer || Buffer.isBuffer(input))) {
+                    throw new TypeError("Expected the `input` argument to be of type `Uint8Array` or `Buffer` or `ArrayBuffer`, got `".concat(typeof input, "`"));
                 }
-            }
-            else if (header[i] !== buffer[i + options.offset]) {
-                return false;
-            }
-        }
-        return true;
-    };
-    var type = file_type_1.default(buffer);
-    if (type == null || (type && type.ext === 'msi')) {
-        // Use CLSIDs to check old Microsoft Office file types: .doc, .xls, .ppt
-        // Ref: http://fileformats.archiveteam.org/wiki/Microsoft_Compound_File
-        var sectorSize = 1 << buffer[30];
-        var index = (buffer[49] * 256) + buffer[48];
-        index = ((index + 1) * sectorSize) + 80;
-        // If the CLSID block is located outside the buffer, it will return an extra field `minimumRequiredBytes`.
-        // Therefore, user can optionally retry it with a larger buffer.
-        if (index + 16 > buffer.length) {
-            return {
-                ext: 'msi',
-                mime: 'application/x-msi',
-                minimumRequiredBytes: index + 16
-            };
-        }
-        // If the CLSID block is located within the buffer, it will try to identify its file type (.doc, .xls, .ppt) by CLSID.
-        if (check([0x06, 0x09, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46], { offset: index })) {
-            return {
-                ext: 'doc',
-                mime: 'application/msword'
-            };
-        }
-        if (check([0x10, 0x08, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46], { offset: index })) {
-            return {
-                ext: 'xls',
-                mime: 'application/vnd.ms-excel'
-            };
-        }
-        if (check([0x20, 0x08, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46], { offset: index })) {
-            return {
-                ext: 'xls',
-                mime: 'application/vnd.ms-excel'
-            };
-        }
-        if (check([0x10, 0x8D, 0x81, 0x64, 0x9B, 0x4F, 0xCF, 0x11, 0x86, 0xEA, 0x00, 0xAA, 0x00, 0xB9, 0x29, 0xE8], { offset: index })) {
-            return {
-                ext: 'ppt',
-                mime: 'application/vnd.ms-powerpoint'
-            };
-        }
-        if (type && type.ext === "msi") {
-            return {
-                ext: 'msi',
-                mime: 'application/x-msi'
-            };
+                buffer = input instanceof Uint8Array ? input : new Uint8Array(input);
+                if (!(buffer && buffer.length > 1)) {
+                    return [2 /*return*/];
```

Comment:
What is this return and yield stuff?

Reply:
I'm guessing it's because we have the tsconfig.json setting the target as es5. Async await is syntactical sugar over the Promise API, and the Promise API itself get's transpiled to generators if the target output version is old enough.

As far as why they are all in comments.... I'm not sure why.

The existing index.js file has a couple of these yields in comments too, just only 2 and now we have 4. Probably related to `fileTypeExt` now being async.